cubic-server/Client.cpp

```diff
@@ -313,6 +321,126 @@ void Client::sendPingResponse(int64_t payload)
     LDEBUG("Sent a ping response");
 }
 
+void Client::sendLoginSuccess(const protocol::LoginSuccess &packet)
+{
+    auto pck = protocol::createLoginSuccess(packet);
+    _sendData(*pck);
+    LDEBUG("Sent a login success");
+
+    switchToPlayState();
+    LDEBUG("Switched to play state");
+
+    protocol::LoginPlay resPck = {
+        .entityID = 0, // TODO: figure out what is this
+        .isHardcore = false, // TODO: something like this this->_player->_dim->getWorld()->getDifficulty();
+        .gamemode = 0, // TODO: something like this this->_player->getGamemode()
+        .previousGamemode = 0, // TODO: something like this this->_player->getPreviousGamemode().has_value() ? this->_player->getPreviousGamemode() : -1;
+        .dimensionCount = 3, // TODO: something like this this->_player->_dim->getWorld()->getDimensions().size();
+        .dimensionNames = std::vector<std::string>({"overworld", "nether", "end"}), // TODO: something like this this->_player->_dim->getWorld()->getDimensions();
+        .registryCodec = std::move(nbt::Compound("registry-codec", {
+            new nbt::Compound("minecraft:dimension_type", {
+                new nbt::String("type", "minecraft:dimension_type"),
+                new nbt::List("value", {
+                    new nbt::Compound("", {
```

Comment:
As you can see there, when nbt's are in a list, they don't have names. Maybe I missunderstood something but I'm quite sure of that
![image](https://user-images.githubusercontent.com/60519602/208810954-a43ea4f1-318a-40db-ba85-3560c024cf99.png)

Reply:
An NBT List contains nameless tags inside of it, that is why it is like this.
I will surely make another constructor at some point to avoid empty strings like this later though.